fpna/DataModel/DataStructure.ipynb

```diff
@@ -0,0 +1,253 @@
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Dimensions"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## AmountType"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record RatioAmountType : AmountType {}",
+                "\npublic record PatternAmountType : AmountType {}",
+                "\npublic record AssetAmountType : AmountType {}",
+                "\npublic record SolvencyBsAmountType : AmountType {}",
+                "\npublic record IfrsBsAmountType : AmountType {}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Cashflow Definition"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record CashflowDefinition : IKeyed",
+                "\n{",
+                "\n    [Key]",
+                "\n    [NotVisible]     ",
+                "\n    public Guid Id { get; init; }",
+                "\n",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(string), nameof(DataNode))]",
+                "\n    public string DataNode { get; init; }",
+                "\n",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(AmountType))] ",
+                "\n    public string AmountType { get; init; }",
+                "\n",
+                "\n    [NoArithmetics(ArithmeticOperation.Scale)]",
+                "\n    [Dimension(typeof(int), nameof(AccidentYear))]",
+                "\n    [IdentityProperty]",
+                "\n    public int? AccidentYear { get; init; }",
+                "\n",
+                "\n    [IdentityProperty]",
+                "\n    [Dimension(typeof(Shape), nameof(Shape))]",
+                "\n    public Shape Shape { get; init; }",
+                "\n       ",
+                "\n    [Dimension(typeof(double), nameof(Values))]",
+                "\n    public double[] Values { get; init; }",
+                "\n",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record SimpleCashflowDefinition : CashflowDefinition",
```

Comment:
This inheritance chain must be thought over. Maybe this is the right option, but I am not sure yet.

Reply:
I make a TODO note in the data model